src/Fixer/StringNotation/HeredocClosingMarkerFixer.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\StringNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class HeredocClosingMarkerFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Unify `heredoc` or `nowdoc` closing marker.',
+            [
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<'TEST'
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['closing_marker' => 'EOF']
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
```

Comment:
```suggestion
                        <?php $a = <<<TEST
```

Code sample should show that `"` is added when `explicit_heredoc_style` is enabled.

Reply:
👍, done.